rdt/transformers/boolean.py

```diff
@@ -20,20 +20,33 @@ class BinaryEncoder(BaseTransformer):
             Indicate what to replace the null values with. If the string ``'mode'`` is given,
             replace them with the most common value.
             Defaults to ``mode``.
+        missing_value_generation (str or None):
+            The way missing values are being handled. There are three strategies:
+
+                * ``RANDOM``: Randomly generates missing values based on the percentage of
+                  missing values.
+                * ``FROM_COLUMN``: Creates a binary column that describes whether the original
+                  value was missing. Then use it to recreate missing values.
+                * ``None``: Do nothing with the missing values on the reverse transform. Simply
+                  pass whatever data we get through.
+
         model_missing_values (bool):
-            Whether to create a new column to indicate which values were null or not. The column
-            will be created only if there are null values. If ``True``, create the new column if
-            there are null values. If ``False``, do not create the new column even if there
-            are null values. Defaults to ``False``.
+            **DEPRECATED** Whether to create a new column to indicate which values were null or
+            not. The column will be created only if there are null values. If ``True``, create
+            the new column if there are null values. If ``False``, do not create the new column
+            even if there are null values. Defaults to ``False``.
     """
 
     INPUT_SDTYPE = 'boolean'
     null_transformer = None
 
-    def __init__(self, missing_value_replacement='mode', model_missing_values=False):
+    def __init__(self, missing_value_replacement='mode', missing_value_generation='RANDOM',
+                 model_missing_values=None):
         super().__init__()
+        self.missing_value_generation = missing_value_generation
```

Comment:
Should we move
`self.missing_value_generation = missing_value_generation` after :
```
if model_missing_values is not None:
            self._set_model_missing_values(model_missing_values)
```
To avoid any wrong re-definition of `missing_value_generation` if the user also uses `model_missing_values`.

Reply:
The reason why first we set `missing_value_generation` is to store (usually) the default value, if a user specifies `model_missing_values` I expect that to dominate over the `default` or `missing_value_generation` and if the user is confused and providing both of them, only the `model_missing_values` will be considered.